lib/std/debug.zig

```diff
@@ -2630,3 +2630,26 @@ pub fn ConfigurableTrace(comptime size: usize, comptime stack_frame_count: usize
         }
     };
 }
+
+pub const SafetyLock = struct {
+    state: State = .unlocked,
+
+    pub const State = if (runtime_safety) enum { unlocked, locked } else enum { unlocked };
```

Comment:
Why not `else void` to not rely on the optimizer? Would this break [@setRuntimeSafety](https://ziglang.org/documentation/master/#toc-setRuntimeSafety)?

Reply:
an enum with only 1 tag is a 0-bit type just like void

Reply:
Today I learned. 😍